buildSrc/src/main/java/org/opensearch/gradle/PublishPlugin.java

```diff
@@ -132,11 +131,10 @@ public String call() throws Exception {
             if (project.getPluginManager().hasPlugin("opensearch.java")) {
                 publication.artifact(project.getTasks().getByName("sourcesJar"));
                 publication.artifact(project.getTasks().getByName("javadocJar"));
+                generatePomTask.configure(
+                    t -> { t.dependsOn(String.format("generatePomFileFor%sPublication", Util.capitalize(publication.getName()))); }
+                );
             }
-
-            generatePomTask.configure(
-                t -> t.dependsOn(String.format("generatePomFileFor%sPublication", Util.capitalize(publication.getName())))
-            );
```

Comment:
Why this change? Feels like the dependency should always be there regardless of what's included in the artifacts.